pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -0,0 +1,257 @@
+package subnetset
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime"
+
+	v1 "k8s.io/api/core/v1"
+	apimachineryruntime "k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/metrics"
+	servicecommon "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/subnet"
+)
+
+var (
+	log                     = logger.Log
+	ResultNormal            = common.ResultNormal
+	ResultRequeue           = common.ResultRequeue
+	ResultRequeueAfter5mins = common.ResultRequeueAfter5mins
+	MetricResTypeSubnetSet  = common.MetricResTypeSubnetSet
+	//TODO rename this
+	defaultSubnet = "default-subnet"
+)
+
+// SubnetSetReconciler reconciles a SubnetSet object
+type SubnetSetReconciler struct {
+	Client  client.Client
+	Scheme  *apimachineryruntime.Scheme
+	Service *subnet.SubnetService
+}
+
+func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
For SubnetSet, we also need to make sure when NS labels changed, update existing subnet tags with new NS labels. this is needed for NetworkPolicy and SecurityPolicy.

Reply:
Yes, I remember that we will do this in following patch?

> For SubnetSet, we also need to make sure when NS labels changed, update existing subnet tags with new NS labels. this is needed for NetworkPolicy and SecurityPolicy.